src/tr/merlinscans/src/eu/kanade/tachiyomi/extension/tr/merlinscans/MerlinScans.kt

```diff
@@ -1,12 +1,451 @@
+                    }
+                }
+
+                is CategoryFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("category", filter.toUriPart())
+                    }
+                }
+
+                is StatusFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("status", filter.toUriPart())
+                    }
+                }
+
+                is SortFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("sort", filter.toUriPart())
+                        sortApplied = true
+                    }
+                }
+
+                else -> {}
+            }
+        }
+
+        if (!sortApplied) {
+            url.addQueryParameter("sort", "views")
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val requestUrl = response.request.url.toString()
+
+        if (requestUrl.contains("search-preview.php")) {
+            val body = response.body.string()
+            return parseSearchJson(body)
+        } else {
+            val document = response.asJsoup()
+            return parseSearchPage(document, response)
+        }
+    }
+
+    private fun parseSearchJson(jsonString: String): MangasPage {
+        val searchResponse = json.decodeFromString<SearchResponse>(jsonString)
+
+        val mangas = if (searchResponse.success) {
+            searchResponse.results.map { result ->
+                SManga.create().apply {
+                    title = result.title
+                    thumbnail_url = result.cover_image
+                    setUrlWithoutDomain(normalizeUrl(result.url))
+
+                    if (result.categories.isNotEmpty()) {
+                        val cleanCategories = result.categories.split(", ")
+                            .distinct()
+                            .take(10)
+                            .joinToString(", ")
+                        description = cleanCategories
+                    }
+                }
+            }
+        } else {
+            emptyList()
+        }
+
+        return MangasPage(mangas, false)
+    }
+
+    private fun parseSearchPage(document: Document, response: Response): MangasPage {
+        val mangas = document.select(searchMangaSelector()).map { searchMangaFromElement(it) }
+        val hasNextPage =
+            searchMangaNextPageSelector()?.let { document.select(it).isNotEmpty() } ?: false
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val normalizedUrl = normalizeUrl(manga.url)
```

Comment:
you already save the normalized urls in db (from browse methods) so why normalize again?